src/textures/DepthTexture.js

```diff
@@ -3,7 +3,7 @@ import { NearestFilter, UnsignedIntType, UnsignedInt248Type, DepthFormat, DepthS
 
 class DepthTexture extends Texture {
 
-	constructor( width, height, type, mapping, wrapS, wrapT, magFilter, minFilter, anisotropy, format ) {
+	constructor( width, height, type, mapping, wrapS, wrapT, magFilter, minFilter, anisotropy, format, compareFunction ) {
```

Comment:
I'm okay with adding `compareFunction` to the ctor signature. However, does it make sense to add a default parameter like for most other texture parameters? 

Maybe `LessEqualCompare` since this would make the creation of the default texture in #27252 nicer.

Reply:
You have written in #27251:

> otherwise we need needsUpdate=true which is probably something we don't want to add in WebGLUniforms.

I don't think this is true. If you do `emptyShadowTexture.compareFunction = LessEqualCompare;`, you don't need to set `needsUpdate` to `true`. 

If we require this PR mainly for #27251, I would try to find a solution without it and maybe think about adding `compareMode` (or a similar name). In this way, the code in `WebGLTextures` could be:
```js
if ( texture.compareMode ) {

    _gl.texParameteri( textureType, _gl.TEXTURE_COMPARE_MODE, _gl.COMPARE_REF_TO_TEXTURE );
    _gl.texParameteri( textureType, _gl.TEXTURE_COMPARE_FUNC, compareToGL[ texture.compareFunction ] );

}
```
I need a closer look at `WebGLState` and `WebGLTextures` if this approach is more in line with the rest of the code base.

Reply:
I think we should stop adding parameters to constructors.
It doesn't scale and ends up with a messy ordering...

Reply:
Okay, that makes sense. In this case, https://github.com/mrdoob/three.js/pull/27251 can set the value of `compareFunction ` as described in https://github.com/mrdoob/three.js/pull/27251#discussion_r1406215208. Meaning:

>  If you do emptyShadowTexture.compareFunction = LessEqualCompare; right after the texture creation, you don't need to set needsUpdate to true.